From 00ca3bd5cf682d95d39f661025b72f45c61d1ab0 Mon Sep 17 00:00:00 2001
From: Neil Chen <Chen.NeilZX@inventec.com>
Date: Mon, 22 Feb 2021 05:37:58 +0000
Subject: [PATCH] [inventec][transformers] KCS3 can't be created because kcs
 binding method is changed. Modify aspeed-g6.dtsi to fix this problem

---
 arch/arm/boot/dts/aspeed-g6.dtsi | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/aspeed-g6.dtsi b/arch/arm/boot/dts/aspeed-g6.dtsi
index da2d8ab..206b665 100644
--- a/arch/arm/boot/dts/aspeed-g6.dtsi
+++ b/arch/arm/boot/dts/aspeed-g6.dtsi
@@ -945,19 +945,22 @@
 						compatible = "aspeed,ast2600-kcs-bmc";
 						interrupts = <GIC_SPI 138 IRQ_TYPE_LEVEL_HIGH>;
 						kcs_chan = <1>;
+						kcs_addr = <0xCA0>;
 						status = "disabled";
 					};
 					kcs2: kcs2@0 {
 						compatible = "aspeed,ast2600-kcs-bmc";
 						interrupts = <GIC_SPI 139 IRQ_TYPE_LEVEL_HIGH>;
 						kcs_chan = <2>;
+						kcs_addr = <0xCA8>;
 						status = "disabled";
 					};
-					kcs3: kcs3@0 {
-						compatible = "aspeed,ast2600-kcs-bmc";
+					kcs3: kcs3@2c {
+						compatible = "aspeed,ast2500-kcs-bmc-v2";
+						reg = <0x2c 0x1>, <0x38 0x1>, <0x44 0x1>;
 						interrupts = <GIC_SPI 140 IRQ_TYPE_LEVEL_HIGH>;
 						kcs_chan = <3>;
-						status = "disabled";
+						aspeed,lpc-io-reg = <0xca2>;
 					};
 				};
 
-- 
2.7.4

